lib/dialects/abstract/query-generator.js

```diff
@@ -2240,6 +2240,11 @@ var QueryGenerator = {
 
       if (escapeValue) {
         value = this.escape(value, field, escapeOptions);
+
+        //if ANY is used with like, add parentheses to generate correct query
+        if (escapeOptions.acceptStrings && (comparator.indexOf('ANY') > comparator.indexOf('LIKE'))) {
+         value = '(' + value + ')';
+        }
```

Comment:
Ideally we still want to avoid string matching, better to add a case for $like/etc with a $any child.